modules/general/key-vault/main.bicep

```diff
@@ -0,0 +1,110 @@
+param enabledForDeployment bool = false
+
+@description('When true, the key vault will be accessible for disk encryption')
+param enabledForDiskEncryption bool = false
+
+@description('When true, the key vault will be accessible by ARM deployments')
+param enabledForTemplateDeployment bool = false
+
+@description('When true, diagnostics settings will be enabled for the key vault')
+param enableDiagnostics bool
+// TODO: Support shipping diagnostics to log analytics?
+
+@description('The storage account name to be used for key vault diagnostic settings')
+param diagnosticsStorageAccountName string = ''
+
+@description('When true, an existing storage account be used for diagnotics settings; When false, the storage account is created/updated')
+param useExistingStorageAccount bool = false
+
+@description('Sets the retention policy for diagnostics settings data, in days')
+param diagnosticsRetentionDays int = 30
+
+@description('When true, the details of an existing key vault will be returned; When false, the key vault is created/udpated')
+param useExisting bool = false
+
+@description('The resource tags applied to resources')
+param resourceTags object = {}
+
+
+targetScope = 'resourceGroup'
+
+
+resource existing_key_vault 'Microsoft.KeyVault/vaults@2021-06-01-preview' existing = if (useExisting) {
+  name: name
+}
+
+resource existing_storage_account 'Microsoft.Storage/storageAccounts@2021-06-01' existing = if (!useExisting && enableDiagnostics && useExistingStorageAccount) {
+  name: diagnosticsStorageAccountName
+}
+module diagnostics_storage '../storage-account/main.bicep' = if (!useExisting && enableDiagnostics && !useExistingStorageAccount) {
+  name: 'kvDiagnosticsDeploy'
+  params: {
+    name: name
+    location: location
+  }
+}
+
+resource key_vault 'Microsoft.KeyVault/vaults@2021-06-01-preview' = if (!useExisting) {
+  name: name
+  location: location
+  properties: {
+    sku: {
+      family: 'A'
+      name: sku
+    }
+    enabledForDeployment: enabledForDeployment
+    enabledForDiskEncryption: enabledForDiskEncryption
+    enabledForTemplateDeployment: enabledForTemplateDeployment
+    tenantId: tenantId
+    accessPolicies: accessPolicies
+
+  }
+  tags: resourceTags
+}
+
+resource keyvault_diagnostics 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (enableDiagnostics) {
+  scope: key_vault
+  name: 'service'
+  properties: {
+    storageAccountId: useExistingStorageAccount ? existing_storage_account.id : diagnostics_storage.outputs.id
+    logs: [
+      {
+        category: 'AuditEvent'
+        enabled: true
+        retentionPolicy: {
+          days: diagnosticsRetentionDays
+          enabled: true
+        }
+      }
+    ]
+  }
+}
+
+
+@description('The resource ID of the key vault')
+output id string = useExisting ? existing_key_vault.id : key_vault.id
+@description('The name of the key vault')
+output name string = useExisting ? existing_key_vault.name : key_vault.name
+
```

Comment:
Is this an extra line, or as in other files you break the outputs up by grouping them with a comment?